### PR TITLE
chore(ref): a bunch of fixes around various things, details in description

### PR DESCRIPTION
**Pull request by Skarlso**
- Fix resource controller double posting to the cache by returning the digest upon creating the resource. This way the resource controller can create that snapshot it wants with the correct digest instead of double pushing to the cache.
- Fix the identity logic which was always using the top level component name as component that resulted in identical identities for all resources. Now, the reconciled component descriptor's name is being used which will be the component that the resource actually belongs to.
- Added AutoDecompress to tackle the problem where the content's compression is outside our reach ( either compressed by our cache, or compressed by OCM, or not compressed at all ( because the layer upon reading decompressed it )
- Fixed `IsCache` using an incorrect format to check if the resource exists in cache.

### CODE REVIEW

File: pkg/ocm/ocm.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type Verifier interface {
 
 // Fetcher gets information about an OCM component Version based on a k8s component Version.
 type Fetcher interface {
-	GetResource(ctx context.Context, cv *v1alpha1.ComponentVersion, resource v1alpha1.ResourceRef) (io.ReadCloser, error)
+	GetResource(ctx context.Context, cv *v1alpha1.ComponentVersion, resource v1alpha1.ResourceRef) (io.ReadCloser, string, error)
```
Comment (Skarlso) [inline pkg/ocm/ocm.go:41]: Digest is needed for the cache to work in the resource controller. Previously it was double posting the content.

File: controllers/mutation_reconcile_looper.go [hunk 2]
```diff
@@ -131,7 +142,13 @@ func (m *MutationReconcileLooper) ReconcileMutationObject(ctx context.Context, s
 	}
 	defer vfs.Cleanup(virtualFS)
 
-	if err := utils.ExtractTarToFs(virtualFS, bytes.NewBuffer(resourceData)); err != nil {
+	extractResourceData, err := Ungzip(bytes.NewBuffer(resourceData))
```
Comment (Skarlso) [inline controllers/mutation_reconcile_looper.go:145]: @phoban01 Let's talk about these! :))

File: controllers/mutation_reconcile_looper.go [hunk 3]
```diff
@@ -131,7 +142,13 @@ func (m *MutationReconcileLooper) ReconcileMutationObject(ctx context.Context, s
...
+	if err != nil {
+		// the data is not compressed
+		extractResourceData = resourceData
```
Comment (Skarlso) [inline controllers/mutation_reconcile_looper.go:148]: This blows my mind. Sometimes it's compressed, sometimes, it isn't.

Comment (Skarlso) [reply controllers/mutation_reconcile_looper.go:148]: Or rather, it's already decompressed and it is a valid TAR file that needs to be uncompressed.

**Comment by Skarlso** [comment]
```
➜  ocm-controller git:(minor-fixes) ✗ k get pods
NAME                       READY   STATUS    RESTARTS   AGE
backend-7ff65895d-4jfjn    1/1     Running   0          2m53s
cache-74b4b54cf-95k8f      1/1     Running   0          2m53s
frontend-c78c6db8d-mhlpv   1/1     Running   0          2m53s
```

File: controllers/configuration_controller_test.go [hunk 4]
```diff
@@ -545,15 +545,15 @@ configuration:
 				require.NoError(t, err)
 				args := cache.PushDataCallingArgumentsOnCall(0)
 				data, name, version := args[0], args[1], args[2]
-				assert.Equal(t, "sha-6558931820223250200", name)
+				assert.Equal(t, "sha-1009814895297045910", name)
 				assert.Equal(t, "999", version)
 
 				t.Log("extracting the passed in data and checking if the configuration worked")
-				dataContent, err := Untar(io.NopCloser(bytes.NewBuffer([]byte(data.(string)))))
+				//dataContent, err := Untar(io.NopCloser(bytes.NewBuffer([]byte(data.(string)))))
```
Comment (Skarlso) [inline controllers/configuration_controller_test.go:552]: I should delete this and the `untar.go` file.

**Review by phoban01 — APPROVED** [review]
LGTM. Good job resolving this decompression issue.

**Pull request merged**